Sources/ComposedUI/CollectionView/CollectionCoordinator.swift

```diff
@@ -192,31 +205,40 @@ open class CollectionCoordinator: NSObject {
         delegate?.coordinatorDidUpdate(self)
     }
 
+    fileprivate func debugLog(_ message: String) {
+        if #available(iOS 12, *) {
+            os_log("%@", log: OSLog(subsystem: "ComposedUI", category: "CollectionCoordinator"), type: .debug, message)
+        }
+    }
 }
 
 // MARK: - SectionProviderMappingDelegate
 
 extension CollectionCoordinator: SectionProviderMappingDelegate {
-
     private func reset() {
-        removes.removeAll()
-        inserts.removeAll()
-        changes.removeAll()
-        moves.removeAll()
-        sectionInserts.removeAll()
-        sectionRemoves.removeAll()
+        debugLog("Resetting")
+        batchedRowRemovals.removeAll()
+        batchedRowInserts.removeAll()
+        batchedRowUpdates.removeAll()
+        batchedRowMoves.removeAll()
+        batchedSectionInserts.removeAll()
+        batchedSectionRemovals.removeAll()
+        batchedSectionUpdates.removeAll()
     }
 
     public func mappingDidInvalidate(_ mapping: SectionProviderMapping) {
         assert(Thread.isMainThread)
+
+        debugLog(#function)
         reset()
         prepareSections()
         collectionView.reloadData()
     }
 
     public func mappingWillBeginUpdating(_ mapping: SectionProviderMapping) {
-        reset()
-        defersUpdate = true
+        batchedUpdatesCount += 1
```

Comment:
Perhaps I'm misunderstanding but for the index stack I suggested, I meant for this value? Not sure how your response relates? Maybe you can elaborate here.

Also, yeah reuse across other coordinators makes total sense and esp from a testing POV.

Reply:
Ah I thought you were referencing the index changes performed in each of the update delegate methods.

Moving `batchedUpdatesCount` to the `Changeset` type mentioned is part of the plan for that

Reply:
Ok cool.